src/Core/Grid/GridFactory.php

```diff
@@ -74,6 +78,10 @@ public function getGrid(SearchCriteriaInterface $searchCriteria)
         $definition = $this->definitionFactory->getDefinition();
         $data = $this->dataFactory->getData($searchCriteria);
 
+        $this->hookDispatcher->dispatchWithParameters('action' . Container::camelize($definition->getId()) . 'GridDataModifier', [
+            'data' => $data,
```

Comment:
`$data` is immutable, isn't it?

Reply:
It should not and be passed by reference. No?

Reply:
make sense @PierreRambaud , ping @sarjon would you mind to complete my pull request, making $data a reference and not a variable?